File: cloud/oauth-server/src/main/java/pers/acp/admin/oauth/constant/OauthApi.java
package pers.acp.admin.oauth.constant;

/**
 * @author zhang by 10/01/2019
 * @since JDK 11
 */
public interface OauthApi {

    String basePath = "/oauth";

    String currUser = "/userinfo";

    String modifiableUser = "/moduserlist";

    String modifiableOrg = "/modorglist";

    String currMenu = "/menulist";

    String appConfig = "/app";

    String updateSecret = appConfig + "/updatesecret";

    String roleConfig = "/role";

    String roleList = roleConfig + "/rolelist";

    String roleCodes = roleConfig + "/rolecodes";

    String authConfig = "/auth";

    String menuConfig = authConfig + "/menu";

    String moduleFuncConfig = authConfig + "/modulefunc";

    String moduleFuncCodes = authConfig + "/modulefunccodes";

    String menuList = authConfig + "/menulist";

    String moduleFuncList = authConfig + "/modulefunclist";

    String userConfig = "/user";

    String userResetPwd = userConfig + "/resetpwd";

    String orgConfig = "/org";

    String runtimeConfig = "/runtime";

    String logOut = "/logout";

    String loginInfo = "/logininfo";

    String onlineInfo = "/onlineinfo";

}

File: cloud/oauth-server/src/main/kotlin/pers/acp/admin/oauth/constant/OauthApi.kt
package pers.acp.admin.oauth.constant

/**
 * @author zhang by 10/01/2019
 * @since JDK 11
 */
object OauthApi {
    const val basePath = "/oauth"
    const val currUser = "/userinfo"
    const val modifiableUser = "/moduserlist"
    const val modifiableOrg = "/modorglist"
    const val currMenu = "/menulist"
    const val appConfig = "/app"
    const val updateSecret = "$appConfig/updatesecret"
    const val roleConfig = "/role"
    const val roleList = "$roleConfig/rolelist"
    const val roleCodes = "$roleConfig/rolecodes"
    const val authConfig = "/auth"
    const val menuConfig = "$authConfig/menu"
    const val moduleFuncConfig = "$authConfig/modulefunc"
    const val moduleFuncCodes = "$authConfig/modulefunccodes"
    const val menuList = "$authConfig/menulist"
    const val moduleFuncList = "$authConfig/modulefunclist"
    const val userConfig = "/user"
    const val userResetPwd = "$userConfig/resetpwd"
    const val orgConfig = "/org"
    const val runtimeConfig = "/runtime"
    const val logOut = "/logout"
    const val onlineInfo = "/onlineinfo"
}